jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/PluginConfigurationProcessor.java

```diff
@@ -122,45 +128,44 @@ static PluginConfigurationProcessor processCommonConfiguration(
             defaultCredentialRetrievers.setKnownCredential(fromCredential, "jib.from.auth"));
     defaultCredentialRetrievers.setCredentialHelperSuffix(jibExtension.getFrom().getCredHelper());
 
-    ImageConfiguration.Builder baseImageConfigurationBuilder =
-        ImageConfiguration.builder(baseImage)
-            .setCredentialRetrievers(defaultCredentialRetrievers.asList());
-
     List<String> entrypoint = computeEntrypoint(logger, jibExtension, projectProperties);
-    ContainerConfiguration.Builder containerConfigurationBuilder =
-        ContainerConfiguration.builder()
+
+    RegistryImage baseImage = RegistryImage.named(baseImageReference);
+    defaultCredentialRetrievers.asList().forEach(baseImage::addCredentialRetriever);
+
+    JibContainerBuilder jibContainerBuilder =
+        Jib.from(baseImage)
+            .setLayers(projectProperties.getJavaLayerConfigurations().getLayerConfigurations())
             .setEntrypoint(entrypoint)
             .setEnvironment(jibExtension.getContainer().getEnvironment())
-            .setProgramArguments(jibExtension.getContainer().getArgs())
             .setExposedPorts(ExposedPortsParser.parse(jibExtension.getContainer().getPorts()))
+            .setProgramArguments(jibExtension.getContainer().getArgs())
             .setLabels(jibExtension.getContainer().getLabels())
             .setUser(jibExtension.getContainer().getUser());
     if (jibExtension.getContainer().getUseCurrentTimestamp()) {
       logger.warn(
           "Setting image creation time to current time; your image may not be reproducible.");
-      containerConfigurationBuilder.setCreationTime(Instant.now());
+      jibContainerBuilder.setCreationTime(Instant.now());
     }
 
-    BuildConfiguration.Builder buildConfigurationBuilder =
-        BuildConfiguration.builder()
-            .setToolName(GradleProjectProperties.TOOL_NAME)
-            .setEventDispatcher(projectProperties.getEventDispatcher())
-            .setAllowInsecureRegistries(jibExtension.getAllowInsecureRegistries())
-            .setLayerConfigurations(
-                projectProperties.getJavaLayerConfigurations().getLayerConfigurations());
-    buildConfigurationBuilder.setApplicationLayersCacheDirectory(
-        projectProperties.getCacheDirectory());
-    buildConfigurationBuilder.setBaseImageLayersCacheDirectory(
-        Containerizer.DEFAULT_BASE_CACHE_DIRECTORY);
-    if (jibExtension.getUseOnlyProjectCache()) {
-      buildConfigurationBuilder.setBaseImageLayersCacheDirectory(
-          projectProperties.getCacheDirectory());
-    }
+    Consumer<Containerizer> containerizerConsumer =
```

Comment:
This seems like the only place where you pass a `containerizerConsumer` to the `PluginConfigurationProcessor`. Do we need to parameterize the consumer? (That is, can't we skip passing a consumer and let `PluginConfigurationProcess.configureContainerizer()` just do the setup there?

Reply:
Yes, I'll move it to a static `configureContainerizer` method. I think the `PluginConfigurationProcessor` will need to be restructured at some point since the logic is a bit unwieldy and interdependent on the plugin goal/task logic.